IdentityCore/src/MSIDExternalCacheSeeder.m

```diff
@@ -0,0 +1,181 @@
+    [tokenRequest sendWithBlock:^(MSIDTokenResponse *tokenResponse, NSError *error)
+     {
+         if (error)
+         {
+             MSID_LOG_ERROR(requestParameters, @"Failed to acquire V1 Id Token token via Refresh token. Error %ld, %@", (long)error.code, error.domain);
+             MSID_LOG_ERROR_PII(requestParameters, @"Failed to acquire V1 Id Token token via Refresh token. Error %@", error);
+             
+             completionBlockWrapper();
+             return;
+         }
+         
+         __auto_type factory = [MSIDAADV2Oauth2FactoryForV1Request new];
+         MSIDIdToken *legacyIdToken = [factory idTokenFromResponse:tokenResponse
+                                                     configuration:requestParameters.msidConfiguration];
+         
+         if (!legacyIdToken)
+         {
+             MSID_LOG_ERROR(requestParameters, @"Failed to parse V1 Id Token. Error %ld, %@", (long)error.code, error.domain);
+             MSID_LOG_ERROR_PII(requestParameters, @"Failed to parse V1 Id Token. Error %@", error);
+             
+             completionBlockWrapper();
+         }
+         
+         MSID_LOG_INFO(requestParameters, @"Saving V1 id token in default cache.");
+         
+         NSError *localError;
+         BOOL result = [self.defaultAccessor saveToken:legacyIdToken context:requestParameters error:&localError];
+         if (result)
+         {
+             MSID_LOG_INFO(requestParameters, @"Saved V1 id token in default cache.");
+         }
+         else
+         {
+             MSID_LOG_ERROR(requestParameters, @"Failed to save V1 id token in default cache. Error %ld, %@", (long)error.code, error.domain);
+             MSID_LOG_ERROR_PII(requestParameters, @"Failed to save V1 id token in default cache. Error %@", error);
+         }
+         
+         [self seedExternalCacheWithIdToken:legacyIdToken
+                              tokenResponse:originalTokenResponse
+                                    factory:factory
+                              configuration:requestParameters.msidConfiguration
+                                    context:requestParameters
+                            completionBlock:completionBlockWrapper];
+     }];
+}
+
+#pragma mark - Private
+
+- (void)seedExternalCacheWithIdToken:(MSIDIdToken *)idToken
+                       tokenResponse:(MSIDTokenResponse *)tokenResponse
+                             factory:(MSIDOauth2Factory *)factory
+                       configuration:(MSIDConfiguration *)configuration
+                             context:(id<MSIDRequestContext>)context
+                     completionBlock:(void(^)(void))completionBlock
+{
+    NSParameterAssert(completionBlock);
+    
+    MSIDLegacyRefreshToken *refreshToken = [factory legacyRefreshTokenFromResponse:tokenResponse
+                                                                     configuration:configuration];
+    refreshToken.idToken = idToken.rawIdToken;
```

Comment:
nit: make sure it sets all the right fields if you set the field afterwards (not sure if there're any fields that are taken from the id token on initialization?)